### DISCUSSION TITLE
is there a way to make op.inObject type safe?

### DISCUSSION THREAD

**Discussion by jo-brue** [cables-gl/cables_docs#922] [Answers & Questions] [answered]
like this opinput always has to have an object of type = {id:string, value: number}

**Comment by steam0r** [comment, marked as answer]
`op.inObject("Audio In", null, "audioNode")` will not let you connect port of different "type" (than "audioNode") to the port, cables will always allow connection port that have no type to ports that have one. 
![Bildschirmfoto 2024-11-20 um 13 44 51](https://github.com/user-attachments/assets/023681ff-fa21-4177-aded-03bac762eb08)

you are suggesting validating the structure of an object, right?
for your custom ops you could decide on some structure and then add "myObject" as a parameter to your own inObject.

**Reply by jo-brue** [reply]
yes thats what i was looking for thanks! 
(Although in my case does not help me since objectGetObject is always of generic type so it still lets me connect any object there but i get it)
